### PR TITLE
add some class roles for linking Errors in doc

### PR DESCRIPTION
**Pull request by fchapoton**
this is adding a few `class` roles so that errors in the doc are linked to python documentation

Here done mostly for `TypeError`

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/structure/parent_old.pyx [hunk 1]
```diff
@@ -221,11 +221,12 @@ cdef class Parent(parent.Parent):
...
         one in turn.  Return the __call__ coercion of the result into
         self of the first canonical coercion that succeeds.  Raise a
-        TypeError if none of them succeed.
+        :class:`TypeError` if none of them succeed.
 
         INPUT:
-             x -- Python object
-             v -- parent object or list (iterator) of parent objects
+
+        - x -- Python object
```
Comment (dcoudert) [inline src/sage/structure/parent_old.pyx:228]: x -> ` ``x`` `

**Review by dcoudert — COMMENTED** [review]
just a minor point. Otherwise, LGTM.

**Comment by fchapoton** [comment]
well, this is inside a deprecated function anyway...

**Review by dcoudert — APPROVED** [review]
OK. So then, I agree with these changes.

**Pull request merged**